### ISSUE TITLE
logger filter is not effective on keys in an array

### ISSUE THREAD

**Issue by ablce9** [hanami/utils#369]
While working with logging sensitive data in arrays, `filter: %w[password]` is not working. 

data comes like this👇 
```console
{ items: [ { password: 'password1'}, { password: 'password2'} ] }
```

Code below from [Hanami logger docs](https://guides.hanamirb.org/projects/logging/)
```ruby
Hanami.configure do
  # ...
  environment :development do
    logger level: :debug, filter: %w[credit_card.number]
  end
end
```
only works for deep hashes. 

```console
[bookshelf] [INFO] [2017-08-11 18:43:04 +0200] HTTP/1.1 PATCH 200 ::1 /billing 2 {"billing"=>{"name"=>"Luca", "address"=>{"street"=>"Centocelle", "number"=>"23", "city"=>"Rome"}, "credit_card"=>{"number"=>"[FILTERED]"}}} 0.009782
```

However, the documentation doesn't really state the case where keys are in _the array_. I do understand a cost it takes to filter out every single key BUT I really appreciate if there is an option for filter the key in arrays.

### What if `credit_card` comes with an array??
`number` should be filtered as expected.
```console
[bookshelf] [INFO] [2017-08-11 18:43:04 +0200] HTTP/1.1 PATCH 200 ::1 /billing 2 {"billing"=>{"name"=>"Luca", "address"=>{"street"=>"Centocelle", "number"=>"23", "city"=>"Rome"}, "credit_cards"=>[{"number"=>"not filtered!!","some_key":"This one should not be filtered"}]}} 0.009782
```
`filter: %w[credit_cards]` generally works but in most cases, another keys exist.

**Comment by emirdaponte** [comment]
Hi, it is true, when build the paths to compare them with the filters, it doesn't contemplate iterating in the arrays. 
https://github.com/hanami/utils/blob/30c158a2fc9c31e0ae88a16b808ac29fdef6fa46/lib/hanami/logger/filter.rb#L43-L47

Maybe it could be contemplated in the following way:
https://github.com/emirdaponte/utils/commit/7ae71be2ceeff8af111426abc4c43d47025e2d8b

What do you think about it?

**Comment by jodosha** [comment]
Fixed by #389
